R/run_stress_test.R

```diff
@@ -36,6 +36,55 @@ run_stress_test <- function(asset_type,
                             shock_year = 2030,
                             term = 2,
                             company_exclusion = TRUE) {
+  warn_if_asset_type_is_longer_than_1(asset_type)
```

Comment:
fallback to default is not a pragmatically useful option for asset type, providing an asset_type of a length > 1 has to result in an error.

Reply:
OK, happy to leave `asset_type` alone. And if the distinction between asset types is so strong, then why not leave it as it was before -- with one interface per type. For example consider the specialized `purrr::map_*()` functions, like `map_int()`, `map_chr()`, etc. You could always use `vapply()` but the extra argument to specify the output tamplate is complex to the `map_*()` family feels friendlier -- particularly when coupled with auto-complete the user has a quick way to see what options there are (like with the enumeration approach) -- without having to read the docs.

Reply:
I think I do not fully understand your suggestion.
Are you suggesting going back to having 1 function per asset type? I would not do that because:

- Conceptually there is no real difference these days between the asset_types (historically they were but we cleaned that up over the last weeks). Remaining differences are mostly due to the fact that we need to handle inconsistencies between the input data. As these are dependent on PACTA results we cannot influence this and have to keep handling the inconsistencies.
- having 3 instead of functions triples copy paste work and testing needs on every change and also bears the danger of reintroducing inconsistencies 
Or are u suggesting providing 3 different interfaces that then call run_stress_test with arg "asset_type" set correctly? That Idea I would be very in favor of, but I do not see how it relates to the map family?

Reply:
> are u suggesting providing 3 different interfaces that then call run_stress_test with arg "asset_type" set correctly?


Yes -- at least to reconsider it. Internally it's always best remove as much duplication as possible. But facing the users it's best to provide whatever it's more user-friendly -- which in this case might be three functions of the same family rather than one function with an additional argument that takes three valid (asset) options.

The relationship between this and the `map_*()` family is that the `map_*()` family could also have been exposed as a single function with an additional argument, but instead it is exposed as three different functions that call the same implementation with three different arguments -- interlally.

``` r
purrr::map_chr
#> function (.x, .f, ...) 
#> {
#>     .f <- as_mapper(.f, ...)
#>     .Call(map_impl, environment(), ".x", ".f", "character")
#> }
#> <bytecode: 0x562907657e18>
#> <environment: namespace:purrr>

purrr::map_int
#> function (.x, .f, ...) 
#> {
#>     .f <- as_mapper(.f, ...)
#>     .Call(map_impl, environment(), ".x", ".f", "integer")
#> }
#> <bytecode: 0x562907055090>
#> <environment: namespace:purrr>

purrr::map_dbl
#> function (.x, .f, ...) 
#> {
#>     .f <- as_mapper(.f, ...)
#>     .Call(map_impl, environment(), ".x", ".f", "double")
#> }
#> <bytecode: 0x5629069c1f48>
#> <environment: namespace:purrr>
```

<sup>Created on 2021-11-03 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>